File: 0228-summary-ranges/0228-summary-ranges.cs
public class Solution {
    public IList<string> SummaryRanges(int[] nums) {
        if (nums == null || nums.Length == 0)
            return new List<string>();
        
        List<string> res = new List<string>();
        int i = 0;
        
        for (int j = 0; j < nums.Length - 1; j++)
            if (nums[j] + 1 != nums[j + 1])
            {
                res.Add(nums[i] == nums[j] ? nums[i].ToString() : nums[i].ToString() + "->" + nums[j].ToString());
                i = j + 1;
            }
        
        res.Add(i == nums.Length - 1 ? nums[i].ToString() : nums[i].ToString() + "->" + nums[nums.Length - 1].ToString());
        
        return res;
    }
}

File: 0228-summary-ranges/0228-summary-ranges.py
from typing import List

class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        if not nums:
            return []
        
        res = []
        start = nums[0]
        
        for i in range(1, len(nums)):
            if nums[i] != nums[i-1] + 1:
                if start == nums[i-1]:
                    res.append(str(start))
                else:
                    res.append(f"{start}->{nums[i-1]}")
                start = nums[i]
        
        if start == nums[-1]:
            res.append(str(start))
        else:
            res.append(f"{start}->{nums[-1]}")
            
        return res
